### ISSUE TITLE
n=5,6 weingarten elements are not orthogonal to character functions

### ISSUE THREAD

**Issue by mduschenes** [polyquantique/haarpy#14]
I seem to see there is a bug with n=5,6 weingarten functions not being orthogonal to their associated character functions for the symmetric group, as below.  Individual weingarten functions appear correct as per your README, and I may be mistaken, thank you!

The code:

```python
from sympy import Symbol,simplify
from sympy.combinatorics.named_groups import SymmetricGroup
from haarpy import weingarten_element

N = [2,3,4,5,6]
d = Symbol('d')

group = lambda n: SymmetricGroup(n).generate_schreier_sims()
character = lambda g,d,n: d**(g.cycles)
weingarten = lambda g,d,n: weingarten_element(g,n,d)

for n in N:
	orthogonality = sum(character(g,d,n)*weingarten(g,d,n) for g in group(n))
	print(n,simplify(orthogonality))
```

yields the following for different n (the inner product of character and weingarten functions over the group should be 1, for all n):

2 1
3 1
4 1
5 (d\*\*5 - 5\*d\*\*3 + 4\*d + 4)/(d*(d\*\*4 - 5\*d\*\*2 + 4))
6 (3\*d\*\*9 - 45\*d\*\*7 + 189\*d\*\*5 + 71\*d\*\*4 - 255\*d\*\*3 - 341\*d\*\*2 + 108\*d + 126)/(3\*d\*(d\*\*8 - 15\*d\*\*6 + 63\*d\*\*4 - 85\*d\*\*2 + 36))

**Comment by nquesada** [comment]
Thanks for the report. We are looking into it. Could you show us a reference where this orthogonality relation is discussed?

Thanks!

**Comment by mduschenes** [comment]
Thanks so much for looking into this! So if you think of the character function $\chi_{d}(\sigma)$ as the $n! \times n!$ graham matrix $G(\sigma,\pi) = \chi_{d}(\sigma^{-1}\pi)$ for all pairs of $d$-dimensional representations of the $n$-order permutations, then for $d>n$, there should exist an inverse of the graham matrix $W = G^{-1}$, the Weingarten matrix $W(\sigma,\pi) = Wg(\sigma^{-1}\pi)$, which should satisfy $WG = I$, or $\sum_{\eta}W(\sigma,\eta)G(\eta,\pi) = \delta_{\sigma\pi}$. I think the issue may be the Weingarten function $Wg(\sigma)$ can also just be defined as the pseudo-inverse of the character, and then the orthogonality with the Graham matrix doesn't hold, but just something up to $GWG = G$. (For $d>n$ (the case I'm personally interested in, in this matrix setting), theinverse/pseudo-inverse definitions should coincide, but maybe this is difficult to implement, and no worries if not?) For example http://arxiv.org/abs/0907.2719. Thanks!

**Comment by nquesada** [comment]
Hi @mduschenes . The bug has been fixed in [#15] . If it is OK with you, we will list you as a contributor to this library.

Thanks for identifying this bug!

**Comment by mduschenes** [comment]
Thanks a lot, so fast to look into it! I don't mind at all being listed, and thanks for the super useful library!
